libs/@local/harpc/net/src/session/server/session_id.rs

```diff
@@ -10,7 +10,12 @@ impl SessionIdProducer {
 
     pub(crate) fn produce(&mut self) -> SessionId {
         let id = self.next;
-        self.next = self.next.wrapping_add(1);
+        self.next += 1;
+
+        // never overflows because right before it reaches usize::MAX, it will be reset to 0
```

Comment:
Could you add a reason why? (Because of the constant below). I think it's good to mention that `MAX` is a reserved special value.